### PR TITLE
CI (Buildkite): use a `TMPDIR` that is backed by real storage

### PR DESCRIPTION
**Pull request by DilumAluthge**

### CODE REVIEW

**Comment by DilumAluthge** [comment]
@staticfloat I am seeing the following test failure:

```
file                               (13) |         failed at 2021-11-08T00:42:22.921
Test Failed at /cache/build/amdci4-5/julialang/julia-master/julia-e6d4b33a81/share/julia/test/file.jl:113
  Expression: dirname(t) == d
   Evaluated: "/cache/build/amdci4-5/julialang/julia-master/tmp" == "/cache/build/amdci4-5/julialang/julia-master/tmp/jl_PnXa94"
```

File: .buildkite/pipelines/main/platforms/tester_linux.yml [hunk 1]
```diff
@@ -61,6 +61,11 @@ steps:
       unset JULIA_DEPOT_PATH
       unset JULIA_PKG_SERVER
 
+      # Make sure that temp files and temp directories are created in a location that is
```
Comment (Keno) [inline .buildkite/pipelines/main/platforms/tester_linux.yml:64]: Why is this required?

Comment (DilumAluthge) [reply .buildkite/pipelines/main/platforms/tester_linux.yml:64]: Inside the Buildkite sandbox, the `/tmp` directory lives on a tempfs that is backed by 1GB of RAM.

On certain Buildkite jobs (e.g. when we run the Pkg test suite with the Pkg server disabled), we end up cloning a lot of Git repos, and as a result we end up running out of space in the tempfs.

Comment (Keno) [reply .buildkite/pipelines/main/platforms/tester_linux.yml:64]: Increase the size of the tempfs?

Comment (DilumAluthge) [reply .buildkite/pipelines/main/platforms/tester_linux.yml:64]: We could do that. I know that @staticfloat and @maleadt have expressed concerns about that in the past. We have multiple Buildkite agents on each machine, and we don't want to use up too much RAM.
